test/inductor/test_distributed_patterns.py

```diff
@@ -146,20 +146,23 @@ def fn(x, obj):
         self.assertEqual(x1.grad, x3.grad)
 
     @torch.no_grad()
+    @requires_gpu()
     def test_storage_resize_zero(self):
         @torch.compile(fullgraph=True)
         def fn(x):
             y = torch.sin(x)
             x.untyped_storage().resize_(0)
             return torch.cos(y)
 
-        x = torch.randn(10)
-        expected = torch.cos(torch.sin(x))
-        y = fn(x)
-        self.assertEqual(y, expected)
-        self.assertEqual(x.untyped_storage().size(), 0)
+        for device in ("cpu", "cuda"):
+            x = torch.randn(10, device=device)
+            expected = torch.cos(torch.sin(x))
+            y = fn(x)
+            self.assertEqual(y, expected)
+            self.assertEqual(x.untyped_storage().size(), 0)
 
     @torch.no_grad()
+    @requires_gpu()
```

Comment:
We should make this work on CPU and CUDA.

Reply:
Updated to split into two tests: CPU and CUDA